hs_core/hydroshare/resource.py

```diff
@@ -751,14 +750,13 @@ def publish_resource(pk):
     Note:  This is different than just giving public access to a resource via access control rul
     """
     resource = utils.get_resource_by_shortkey(pk)
-    resource.published_and_frozen = True
-    resource.frozen = True
-    resource.edit_users = []
-    resource.edit_groups = []
+    resource.raccess.published = True
```

Comment:
This is correct. Setting a resource to be immutable overrides the edit_groups and edit_users settings.